cloudinit/util.py

```diff
@@ -1724,6 +1753,8 @@ def get_output_cfg(cfg, mode) -> List[Optional[str]]:
     if mode in outcfg:
         modecfg = outcfg[mode]
     else:
+        # TODO: This makes no sense. If they ask for "junk" mode we give
+        # them back "all" if it exists?
```

Comment:
+1 agreed - a warning in this case would probably be appropriate

Reply:
I think this is actually the default case :smile: 

If a mode isn't specified, we hit this else, so I'm not sure a warning is appropriate.